crates/walrus-service/src/node.rs

```diff
@@ -2835,53 +2973,100 @@ impl ServiceState for StorageNodeInner {
     async fn store_metadata(
         &self,
         metadata: UnverifiedBlobMetadataWithId,
+        intent: UploadIntent,
     ) -> Result<bool, StoreMetadataError> {
-        let Some(blob_info) = self
+        let blob_id = *metadata.blob_id();
+
+        if self
             .storage
-            .get_blob_info(metadata.blob_id())
-            .context("could not retrieve blob info")?
-        else {
-            return Err(StoreMetadataError::NotCurrentlyRegistered);
-        };
+            .has_metadata(&blob_id)
+            .context("could not check metadata existence")?
+        {
+            return Ok(false);
+        }
```

Comment:
Suggestion: As the function `has_metadata` internally also gets the blob info, I'd suggest to keep the code to attempt getting the blob info here (instead of a few lines below) and perform further checks on it.

Reply:
done